### PR TITLE
Change to use private features to allow toleration of resteasy cdi ee…

### LINKED ISSUE #21969: Add EE10 toleration to mpRestClient-3.0
mpRestClient-3.0 is currently based on restfulWS-3.0 (EE9). We need to add toleration of EE10 which means allowing the feature to be based on restfulWS-3.1.

### PR DESCRIPTION
**Pull request by jim-krueger**
Change to use private features to allow toleration of resteasy cdi ee9 and ee10 versions

fixes #21969

### CODE REVIEW

**Comment by jim-krueger** [comment]
#run-libby-bot

**Comment by jim-krueger** [comment]
#build

**Comment by jim-krueger** [comment]
fixes #21969

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.mpRestClient.internal.cdi-3.0.feature [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=io.openliberty.mpRestClient.internal.cdi-3.0
+singleton=true
+visibility=private
+-bundles=io.openliberty.org.jboss.resteasy.cdi; apiJar=false; location:="lib/"
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.mpRestClient.internal.cdi-3.0.feature]:
```suggestion
-bundles=io.openliberty.org.jboss.resteasy.cdi
```
apiJar=false and location:="lib" are not needed here.

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.mpRestClient.internal.cdi-4.0.feature [hunk 2]
```diff
@@ -0,0 +1,7 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=io.openliberty.mpRestClient.internal.cdi-4.0
+singleton=true
+visibility=private
+-bundles=io.openliberty.org.jboss.resteasy.cdi.ee10; apiJar=false; location:="lib/"
+kind=ga
+edition=core
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.mpRestClient.internal.cdi-4.0.feature]:
```suggestion
-bundles=io.openliberty.org.jboss.resteasy.cdi.ee10
kind=noship
edition=full
```

**Review by jhanders34 — CHANGES_REQUESTED** [review]

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.mpRestClient.internal.cdi-3.0.feature [hunk 3]
```diff
@@ -0,0 +1,7 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=io.openliberty.mpRestClient.internal.cdi-3.0
+singleton=true
+visibility=private
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.mpRestClient.internal.cdi-3.0.feature:4]:
```suggestion
visibility=private
-features=io.openliberty.restfulWSClient-3.0
```

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.mpRestClient.internal.cdi-4.0.feature [hunk 4]
```diff
@@ -0,0 +1,7 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=io.openliberty.mpRestClient.internal.cdi-4.0
+singleton=true
+visibility=private
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.mpRestClient.internal.cdi-4.0.feature:4]:
```suggestion
visibility=private
-features=io.openliberty.restfulWSClient-3.1
```

File: dev/com.ibm.websphere.appserver.features/visibility/public/mpRestClient-3.0/io.openliberty.mpRestClient-3.0.feature [hunk 5]
```diff
@@ -22,11 +22,11 @@ Subsystem-Name: MicroProfile Rest Client 3.0
   io.openliberty.restfulWSClient-3.0, \
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/public/mpRestClient-3.0/io.openliberty.mpRestClient-3.0.feature]:
```suggestion
  io.openliberty.restfulWSClient-3.0; ibm.tolerates:="3.1", \
```

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.mpRestClient.internal.cdi-3.0.feature (hunk 1 above)
Comment (jim-krueger) [reply dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.mpRestClient.internal.cdi-3.0.feature]: Done

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.mpRestClient.internal.cdi-4.0.feature (hunk 2 above)
Comment (jim-krueger) [reply dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.mpRestClient.internal.cdi-4.0.feature]: Done

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.mpRestClient.internal.cdi-3.0.feature (hunk 3 above)
Comment (jim-krueger) [reply dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.mpRestClient.internal.cdi-3.0.feature:4]: Done

File: dev/com.ibm.websphere.appserver.features/visibility/public/mpRestClient-3.0/io.openliberty.mpRestClient-3.0.feature (hunk 5 above)
Comment (tevans78) [reply dev/com.ibm.websphere.appserver.features/visibility/public/mpRestClient-3.0/io.openliberty.mpRestClient-3.0.feature]: Adding toleration of 3.1 is good but won't work by itself. Using version 3.1 changes the version of resteasy being used and that then requires updates to the `io.openliberty.org.jboss.resteasy.mprestclient` bundle.

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.mpRestClient.internal.cdi-4.0.feature (hunk 4 above)
Comment (jim-krueger) [reply dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.mpRestClient.internal.cdi-4.0.feature:4]: Done.

File: dev/com.ibm.websphere.appserver.features/visibility/public/mpRestClient-3.0/io.openliberty.mpRestClient-3.0.feature (hunk 5 above)
Comment (jim-krueger) [reply dev/com.ibm.websphere.appserver.features/visibility/public/mpRestClient-3.0/io.openliberty.mpRestClient-3.0.feature]: Done

**Comment by jim-krueger** [comment]
#run-libby-bot

**Comment by jim-krueger** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/public/mpRestClient-3.0/io.openliberty.mpRestClient-3.0.feature (hunk 5 above)
Comment (jhanders34) [reply dev/com.ibm.websphere.appserver.features/visibility/public/mpRestClient-3.0/io.openliberty.mpRestClient-3.0.feature]: Need a combination of Tom and Jim's changes to really test this with EE10.  The tolerates is needed so we pick the right private feature which is why I added that to the review.  Tom's changes will add more stuff.  So Jim you may need to make some updates for the mprestclient bundle if it fails to load due to 6.1.0 changes, but the hope is that it wouldn't, but until you test with EE10 features you won't know that.

Comment (jim-krueger) [reply dev/com.ibm.websphere.appserver.features/visibility/public/mpRestClient-3.0/io.openliberty.mpRestClient-3.0.feature]: Hadn't seen Tom's comments prior to marking this as "done".

**Comment by jim-krueger** [comment]
#run-libby-bot

**Comment by jim-krueger** [comment]
#build

**Review by jhanders34 — APPROVED** [review]

**Review by WhiteCat22 — APPROVED** [review]

**Pull request merged**